packages/date/src/index.js

```diff
@@ -440,4 +463,49 @@ export function getDate( dateString ) {
 	return momentLib.tz( dateString, WP_ZONE ).toDate();
 }
 
+/**
+ * Creates a moment instance using the given timezone or, if none is provided, using global settings.
+ *
+ * @param {(Date|string|Moment|null)}   dateValue  Date object or string, parsable
+ *                                                 by moment.js.
+ * @param {(string|number|null)}        timezone   Timezone to output result in or
+ *                                                 a UTC offset.
+ *                                                 Defaults to timezone from site.
+ *                                                 See momentjs.com.
+ * @return {Moment}  a moment instance.
+ */
+function buildMoment( dateValue, timezone ) {
+	if ( timezone && ! isUtcOffset( timezone ) ) {
+		return momentLib( dateValue ).tz( timezone );
+	}
+
+	if ( timezone && isUtcOffset( timezone ) ) {
+		return momentLib( dateValue ).utcOffset( timezone );
+	}
+
+	if ( settings.timezone.string ) {
+		return momentLib( dateValue ).tz( settings.timezone.string );
+	}
+
+	return momentLib( dateValue ).utcOffset( settings.timezone.offset );
+}
+
+const VALID_UTC_OFFSET = /^[+-][0-1][0-9](:[0-9][0-9])?$/;
+
+/**
+ * Returns whether a certain UTC offset is valid or not.
+ *
+ * @param {number|string} offset a UTC offset.
+ *
+ * @return {boolean} whether a certain UTC offset is valid or not.
+ */
+function isUtcOffset( offset ) {
```

Comment:
Minor (since it's internal at least): Correct capitalization would be:

```suggestion
function isUTCOffset( offset ) {
```

Reference: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#abbreviations-and-acronyms

Reply:
Fixed in d914dbb.